### PR TITLE
Feature/lib test cases

### PR DESCRIPTION
**Pull request by aliasgharkarani**
Test Cases created for 
1) RNColorPlalette 
2) Color-picker
3) Color-wheel

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -27,6 +27,23 @@
   },
   "homepage": "https://github.com/IOMechs/RNColorPalette#readme",
   "devDependencies": {
-    "metro-react-native-babel-preset": "^0.58.0"
+    "@babel/core": "^7.1.2",
```
Comment (AhsanAyaz) [inline package.json]: @aliasgharkarani you sure all of these packages are required for testing? I.e. we don't have to remove any of these.

File: __tests__/RNColorPlalette-test.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import RNColorPalette from '../src/RNColorPalette';
+import renderer from 'react-test-renderer';
+
+test('renders correctly', () => {
+  const tree = renderer.create(<RNColorPalette />).toJSON();
+  expect(tree).toMatchSnapshot();
+});
+
+
+it('togglePalette', () => {
```
Comment (AhsanAyaz) [inline __tests__/RNColorPlalette-test.js]: The description of the test inside `it` should be very clear on what it should do.
For instance
`it('should change the state of the color picker when we toggle the palette', () => {})`

File: __tests__/__snapshots__/RNColorPlalette-test.js.snap [hunk 3]
```diff
@@ -0,0 +1,307 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
```
Comment (AhsanAyaz) [inline __tests__/__snapshots__/RNColorPlalette-test.js.snap]: Do we need to commit these snapshots? Just curious.

**Comment by AhsanAyaz** [comment]
@aliasgharkarani , this file `rn-color-palette-1.0.0.tgz` is not in the repository.
How are you able to run the tests as I'm not able to do so locally.
Also, in the demo, we should be using the latest package (from artifactory). So you should install the actual package i.e. `@iomechs/rn-color-palette` from npm in demo. And then test the tests.

File: __tests__/__snapshots__/RNColorPlalette-test.js.snap (hunk 3 above)
Comment (aliasgharkarani) [reply __tests__/__snapshots__/RNColorPlalette-test.js.snap]: removed.

File: __tests__/RNColorPlalette-test.js (hunk 2 above)
Comment (aliasgharkarani) [reply __tests__/RNColorPlalette-test.js]: description added.

File: package.json (hunk 1 above)
Comment (aliasgharkarani) [reply package.json]: extra packages removed.

**Pull request merged**